lib/resty/core/regex.lua

```diff
@@ -478,6 +479,84 @@ function ngx.re.find(subj, regex, opts, ctx, nth)
     return re_match_helper(subj, regex, opts, ctx, false, nil, nth)
 end
 
+local function destroy_re_gmatch_iterator(iterator)
+    if not iterator._compile_once then
+        destroy_compiled_regex(iterator._compiled)
```

Comment:
Better clear the `iterator._compiled` field with `nil` here as well? Oh, and other fields too.

Reply:
Wouldn't the `iterator` table be GC'ed anyways, and so will `._compiled` be too? I am curious to hear more about this. Does it help because `._compiled` is a cdata object?

I have two suggestions here:
* if we want to unset all fields of the `iterator` table, then only 3 fields are currently unset. Should we unset them all? Or do we only want to clear cdata objects?
* we could add a test asserting that an iterator that error-ed out returns `nil` on subsequent calls (Thus effectively testing that we unset `._pos` here, as we should)

Reply:
@thibaultcha This serves as a minor optimization by releasing the GC references sooner rather than later.

Reply:
Gotcha. That was my hunch as well. Then I think we can avoid the `._subj = nil` and just unset `._compiled` and `._pos` maybe. Plus the mentioned test - I believe it'd be a healthy one to have.

Reply:
@thibaultcha Either way is fine :)

Reply:
@thibaultcha 
`._subj = nil` frees the reference to the `subj`, so it could be GCed eariler.